espnet2/enh/espnet_enh_s2t_model.py

```diff
@@ -329,6 +401,97 @@ def nll(
 
     batchify_nll = ESPnetASRModel.batchify_nll
 
+    def asr_pit_loss(self, speech, speech_lengths, text, text_lengths):
```

Comment:
I have a rough idea about implementing all the pit loss within a PIT class. Maybe we can implement this later. In this PIT class, it can compute the pit loss for all criterions and determine the optimal permutation, can reorder the data, etc. The major target is to keep the code minimal and clean.

Reply:
Sure.  Indeed, this part can be wrapped as a standalone module. We can add support for pit later, and for now we can always use `enh_model` to determine the permutation.

Reply:
Related to another comment from @simpleoier, can we use the permutation defined by `asr_pit_loss` for the enhancement loss?

Reply:
> can we use the permutation defined by asr_pit_loss for the enhancement loss?

This is possible. But the implementation might be complex. We can consider adding this function later.

Reply:
I see, we can do it later.